daemon/conman/connection.go

```diff
@@ -107,26 +128,18 @@ func newConnectionImpl(nfp *netfilter.Packet, c *Connection, protoType string) (
 		c.Entry.UserId = int(nfp.UID)
```

Comment:
@gustavo-iniguez-goya , do you know what this line is for
> else if c.Entry.UserId == -1 && nfp.UID != 0xffffffff

more specifically, why can we not rely on UID reported from netfilter?

Reply:
in some systems libnetfilter_queue doesn't obtain the UID (<= 1.0.3 if I remember correctly).

Reply:
moreover, some connections inititated from a container doesn't report it:
new connection udp => 41809:172.17.0.5 -> 208.67.220.220:53 uid: 4294967295

probably because going through the FORWARD table.

Reply:
ok, I'll add it as a comment to the code then.

Reply:
So, is it fair to say that what the code does is - if it fails to find the UID, it falls back to using the UID reported via netfilter_queue, even though that UID potentially may be incorrect ?

Reply:
yes, I think so. We'll run into this scenario if we add a FORWARD rule to intercept containerized connections (it's working fine as far as I can tell). In the near future would be great if we could display the container name.
But if an external connection to the box traverses the FORWARD table then we won't get the UID.